### ISSUE TITLE
TypeError: Cannot set property 'error' of undefined

### ISSUE THREAD

**Issue by jeyarajcs** [DataDog/dd-trace-js#2235]
I am getting infinite error logs from dd-trace If the code throws any exception.

App.js
```
const tracer = require('dd-trace');
if(process.env.TRACE_REQUIRED === "true"){
    tracer.init();
    tracer.use('http', {
        blocklist: ['/health', '/ping']
    })
}
```

Express.js middleware function:
```
function checkUser(req, res){
		let user = await db.users.findOne({_id: "xyz"});
		if(!user) {
			errorUtils.throwError('User not found', 400);
		}
		req.user = user;
	}

	next();
}
```

errorUtils.js
```
exports.throwError = throwError;
function throwError(message, code) {
    let error = new Error(message);
    if (code) {
        error.code = code;
    }
    throw error;
}
```

When there is no user found in the MongoDB query, dd-trace is started throwing lot error logs infinitely like `Cannot set property 'error' of undefined` etc.
The problem is not only in the middleware function, but wherever I am using the `errorUtils.throwError('XYZ', 400);` I am getting the same errors.

NodeJS version : 12.x
dd-trace version: tried 2.8.0 as well as 2.12.0 But same errors.

Could anyone help me to resolve this?

![image](https://user-images.githubusercontent.com/13285081/181739671-4827db84-a20f-4702-b524-6b7facb0d335.png)

**Comment by rochdev** [comment]
I opened #2240 to fix the error. However, this would only happen when an HTTP span is not active when the error occurs, which would indicate there is also another issue. This would usually happen when the tracer is not initialized before instrumented modules are imported. Is the tracer initialization the first thing happening in the process?

**Comment by jeyarajcs** [comment]
Yes.. The tracer initialization is the first thing at my code.

I checked the file that showed in the error log `/node_modules/dd-trace/packages/dd-trace/src/plugins/util/web.js:273:21` and its pointed to the `addError` function.

```
  addError (req, error) {
    if (error instanceof Error) {
      const context = contexts.get(req)
      context.error = error
    }
  },
  ```
  
  I just added an extra `if` condition and tested locally(at my laptop), No more `TypeError: Cannot set property 'error' of undefined` errors.
  ```
    addError (req, error) {
    if (error instanceof Error) {
      const context = contexts.get(req)
      if(context){
        context.error = error
      }
      
    }
  },
  ```

**Comment by rochdev** [comment]
It looks like this would also happen if the context is lost in the middleware, which would end up hiding the error completely from the tracer. I'll work on a fix to not only fix the exception but also make sure that the error is captured.

**Comment by jeyarajcs** [comment]
Perfect.
Will this fix be pushed any time soon?
Actually, We found this from our production system, and temporarily we stopped the tracing.

**Comment by rochdev** [comment]
I will publish a new release as soon as I'm done with the fix, hopefully by end of day.

**Comment by rochdev** [comment]
Fixed in 2.12.1
